lib/ContractInteract/TokenRules.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+const Web3 = require('web3');
+const AbiBinProvider = require('./../AbiBinProvider');
+const Contracts = require('../Contracts');
+
+const Utils = require('../../utils/Utils');
+const ContractName = 'TokenRules';
+
+/**
+ * This library is used to interact with Recovery contracts.
+ */
+class TokenRules {
+  /**
+   * TokenRules class constructor.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary web3 object.
+   * @param {String} address delayedRecovery proxy contract address.
+   */
+  constructor(auxiliaryWeb3, address) {
+    if (!(auxiliaryWeb3 instanceof Web3)) {
+      const err = new TypeError("Mandatory Parameter 'auxiliaryWeb3' is missing or invalid.");
+      return Promise.reject(err);
+    }
+
+    if (!Web3.utils.isAddress(address)) {
+      const err = new TypeError(`Mandatory Parameter 'address' is missing or invalid: ${address}.`);
+      return Promise.reject(err);
+    }
+
+    this.auxiliaryWeb3 = auxiliaryWeb3;
+    this.address = address;
+
+    this.contract = Contracts.getTokenRules(this.auxiliaryWeb3, this.address);
+
+    if (!this.contract) {
+      const err = new TypeError(`Could not load TokenRules contract for: ${this.address}`);
+      return Promise.reject(err);
+    }
+  }
+
+  /**
+   * Deploys TokenRules contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Origin chain web3 object.
+   * @param {String} organization Organization which holds all the keys needed to administer the economy.
+   * @param {String } eip20Token EIP20 token contract address deployed for an economy.
+   * @param {Object} txOptions Tx options.
+   *
+   * @returns {Promise<TokenRules>} Promise containing the TokenRules
+   *                              instance that has been deployed.
+   */
+  static async deploy(auxiliaryWeb3, organization, eip20Token, txOptions) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    if (!Web3.utils.isAddress(txOptions.from)) {
+      const err = new TypeError(`Invalid from address: ${txOptions.from}.`);
+      return Promise.reject(err);
+    }
+
+    const tx = TokenRules.deployRawTx(auxiliaryWeb3, organization, eip20Token);
+
+    return Utils.sendTransaction(tx, txOptions).then((txReceipt) => {
+      const address = txReceipt.contractAddress;
+      return new TokenRules(auxiliaryWeb3, address);
+    });
+  }
+
+  /**
+   * Method which returns Tx object to deploy TokenRules contract.
+   *
+   * @param {Web3} auxiliaryWeb3 Auxiliary chain web3 object.
+   *
+   * @returns {Object} Raw transaction object.
+   */
+  static deployRawTx(auxiliaryWeb3, organization, eip20Token) {
```

Comment:
Please fix the method documentation

Reply:
Also add validation for a valid address.

Reply:
Done!